### PR TITLE
fix: localtime error

### PR DESCRIPTION
**Pull request by fuchengjie**
when environment is win10 and vs2019
i encountered this problem
image

And add this macro can solve it.

### CODE REVIEW

**Comment by fuchengjie** [comment]
@sharkdp，check it pls

**Comment by sharkdp** [comment]
I'm not sure this is the correct way to fix this problem (silencing the warning). How about applying an actual fix (switch to `localtime_s`)?

**Comment by fuchengjie** [comment]
I’ll try it

**Comment by fuchengjie** [comment]
I will try it

**Comment by fuchengjie** [comment]
@sharkdp, i use localtime_s to fix error, pls check it.

File: dbg.h [hunk 1]
```diff
@@ -550,7 +550,12 @@ inline bool pretty_print(std::ostream& stream, const time&) {
   const auto us =
       duration_cast<microseconds>(now.time_since_epoch()).count() % 1000000;
   const auto hms = system_clock::to_time_t(now);
+#if _MSC_VER >= 1600
+  const auto tm = new std::tm;
```
Comment (sharkdp) [inline dbg.h:554]: This allocates a new `tm` object on the heap and never releases the memory.

File: dbg.h [hunk 2]
```diff
@@ -550,7 +550,12 @@ inline bool pretty_print(std::ostream& stream, const time&) {
   const auto us =
       duration_cast<microseconds>(now.time_since_epoch()).count() % 1000000;
   const auto hms = system_clock::to_time_t(now);
+#if _MSC_VER >= 1600
```
Comment (sharkdp) [inline dbg.h:553]: Why do we want to do this for `_MSC_VER >= 1600` only and not everywhere?

Comment (fuchengjie) [reply dbg.h:553]: localtime_s was introduced in vs2010(_MSC_VER >= 1600), so just replace localtime with localtime_s here.

File: dbg.h [hunk 3]
```diff
@@ -550,10 +550,15 @@ inline bool pretty_print(std::ostream& stream, const time&) {
...
+  const auto tm = new std::tm;
+  localtime_s(tm, &hms);
+#else
   const std::tm* tm = std::localtime(&hms);
+#endif
   stream << "current time = " << std::put_time(tm, "%H:%M:%S") << '.'
          << std::setw(6) << std::setfill('0') << us;
-
+  delete tm;
```
Comment (sharkdp) [inline dbg.h:561]: Now you are calling `delete` on a `tm*` pointer to a static `tm` object inside `std::localtime` when the `#else` branch is active. Please don't use `new`/`delete` at all here but rather simply "stack-allocate" a `tm` object locally.

File: dbg.h (hunk 2 above)
Comment (sharkdp) [reply dbg.h:553]: Ok, and how about non-MSV environments?

Comment (fuchengjie) [reply dbg.h:553]: non-MSV environments also used old #else statements. 
added macro just fix MSV environments error. /Sa my first comment.

**Comment by sharkdp** [comment]
Thank you very much!

**Comment by fuchengjie** [comment]
> Thank you very much!

It's my pleasure.

**Pull request merged**